### PR TITLE
Add citation section in README.md

### PR DESCRIPTION
**Pull request by WoutLegiest**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,19 @@ There are many ways to contribute to HEIR:
...
     HEIR.
 
+## Citations
+
+The HEIR project can be cited in in academic work through following entry:
+
+```text
+@Misc{HEIR,
+  title={{HEIR: Homomorphic Encryption Intermediate Representation}},
+  author={Google LLC},
```
Comment (AlexanderViand-Intel) [inline README.md:70]: The convention for open-source projects is usually something like "<project> contributors"
```suggestion
  author={HEIR Contributors},
```

**Review by AlexanderViand-Intel — APPROVED** [review]
Great idea 👍 This reminds that we should also start thinking about plans for something more "indexable" (i.e., something with a DOI) so we can keep track of citations!

**Comment by WoutLegiest** [comment]
> Great idea 👍 This reminds that we should also start thinking about plans for something more "indexable" (i.e., something with a DOI) so we can keep track of citations!

We were thinking the same!

**Review by j2kun — APPROVED** [review]
Please squash into one commit, thanks!

**Pull request merged**